src/main/java/org/sonatype/nexus/ci/iq/IqPolicyEvaluatorUtil.groovy

```diff
@@ -49,8 +51,20 @@ class IqPolicyEvaluatorUtil
           new IqClientFactoryConfiguration(credentialsId: iqPolicyEvaluator.jobCredentialsId, context: run.parent,
               log: loggerBridge))
 
-      def verified = iqClient.verifyOrCreateApplication(applicationId)
-      checkArgument(verified, 'The application ID ' + applicationId + ' is invalid.')
+      try {
+        def verified = iqClient.verifyOrCreateApplication(applicationId)
+        checkArgument(verified, 'The application ID ' + applicationId + ' is invalid.')
+      }
+      catch (IqClientException e) {
+        if (isHttp404Error(e)) {
+          listener.logger.println Messages._IqPolicyEvaluation_AutoAppMissing()
+          run.result = Result.FAILURE
+          return null
```

Comment:
What happens after you return null here? Isn't that still an exception that must be thrown?

Reply:
> What happens after you return null here? Isn't that still an exception that must be thrown?

No exception is thrown.  I inform Jenkins it failed by setting the ``run.result = Result.FAILURE```.   If I did throw an exception it will be displayed instead of my message.   If I let the exception be thrown it would change the output to be the stacktrace.

Reply:
> No exception is thrown. I inform Jenkins it failed by setting the run.result = Result.FAILURE. If I did throw an exception it will be displayed instead of my message. If I let the exception be thrown it would change the output to be the stacktrace.

Why is that undesired? Why not throw an exception with the new message and the cause set to the caught exception?